ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -152,9 +155,10 @@ public static String getInstanceName(String host, Integer port) {
       String name = entry.getString(DATACENTER_STR);
       byte id = (byte) entry.getInt(DATACENTER_ID_STR);
       ReplicaType replicaType = entry.optEnum(ReplicaType.class, REPLICA_TYPE_STR, ReplicaType.DISK_BACKED);
-      String zkConnectStr = (replicaType == ReplicaType.DISK_BACKED) ? entry.getString(ZKCONNECTSTR_STR)
-          : entry.optString(ZKCONNECTSTR_STR);
-      DcZkInfo dcZkInfo = new DcZkInfo(name, id, zkConnectStr, replicaType);
+      String[] zkConnectStrs =
+          (replicaType == ReplicaType.DISK_BACKED) ? entry.getString(ZKCONNECTSTR_STR).split(ZKCONNECTSTR_DELIMITER)
+              : entry.optString(ZKCONNECTSTR_STR).split(ZKCONNECTSTR_DELIMITER);
```

Comment:
this can result in a NPE, since we use optString for cloud backed replicas.

Instead, you could do `Utils.splitString(entry.optString(ZKCONNECT_STR, ""), ZKCONNECTSTR_DELIMITER)`

Reply:
Yeah, `Utils.splitString` is better. I previously used `split` with the impression that `optString` returns empty `""` if not found, which doesn't cause NPE. Anyway, I will go with `Utils.splitString`.

Reply:
Oh interesting, you are right about it returning empty string instead of null:
```
    public String optString(String key) {
        return this.optString(key, "");
    }
```
You can ignore the comment about the NPE. Still it will be good to use Utils.splitString since it will filter out the empty string. i.e. return the list `[]` instead of `[""]`